File: Dddml.Wms.HttpServices.PhpClient/Dddml/Wms/Domain/MonthPlanMvoStateEventIdFlattenedDto.php
<?php

namespace Dddml\Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\FlattenedDtoInterface;

class MonthPlanMvoStateEventIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'monthPlanIdPersonalNameFirstName' => 'string',
            'monthPlanIdPersonalNameLastName' => 'string',
            'monthPlanIdYear' => 'integer',
            'monthPlanIdMonth' => 'integer',
            'personVersion' => 'Long',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var MonthPlanMvoStateEventId
     */
    private $value;

    /**
     * @param MonthPlanMvoStateEventId $value
     */
    public function __construct(MonthPlanMvoStateEventId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new MonthPlanMvoStateEventId();
        }
    }

    /**
     * @return MonthPlanMvoStateEventId
     */
    public function toMonthPlanMvoStateEventId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getMonthPlanIdPersonalNameFirstName()
    {
        return $this->value->getMonthPlanId()->getPersonalName()->getFirstName();
    }

    /**
     * @param string $monthPlanIdPersonalNameFirstName
     */
    public function setMonthPlanIdPersonalNameFirstName($monthPlanIdPersonalNameFirstName)
    {
        $this->value->getMonthPlanId()->getPersonalName()->setFirstName($monthPlanIdPersonalNameFirstName);
    }

    /**
     * @return string
     */
    public function getMonthPlanIdPersonalNameLastName()
    {
        return $this->value->getMonthPlanId()->getPersonalName()->getLastName();
    }

    /**
     * @param string $monthPlanIdPersonalNameLastName
     */
    public function setMonthPlanIdPersonalNameLastName($monthPlanIdPersonalNameLastName)
    {
        $this->value->getMonthPlanId()->getPersonalName()->setLastName($monthPlanIdPersonalNameLastName);
    }

    /**
     * @return integer
     */
    public function getMonthPlanIdYear()
    {
        return $this->value->getMonthPlanId()->getYear();
    }

    /**
     * @param integer $monthPlanIdYear
     */
    public function setMonthPlanIdYear($monthPlanIdYear)
    {
        $this->value->getMonthPlanId()->setYear($monthPlanIdYear);
    }

    /**
     * @return integer
     */
    public function getMonthPlanIdMonth()
    {
        return $this->value->getMonthPlanId()->getMonth();
    }

    /**
     * @param integer $monthPlanIdMonth
     */
    public function setMonthPlanIdMonth($monthPlanIdMonth)
    {
        $this->value->getMonthPlanId()->setMonth($monthPlanIdMonth);
    }

    /**
     * @return Long
     */
    public function getPersonVersion()
    {
        return $this->value->getPersonVersion();
    }

    /**
     * @param Long $personVersion
     */
    public function setPersonVersion($personVersion)
    {
        $this->value->setPersonVersion($personVersion);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getMonthPlanIdPersonalNameFirstName(),
            $this->getMonthPlanIdPersonalNameLastName(),
            $this->getMonthPlanIdYear(),
            $this->getMonthPlanIdMonth(),
            $this->getPersonVersion(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return MonthPlanMvoStateEventIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setMonthPlanIdPersonalNameFirstName($pValues[0]);
        $this->setMonthPlanIdPersonalNameLastName($pValues[1]);
        $this->setMonthPlanIdYear($pValues[2]);
        $this->setMonthPlanIdMonth($pValues[3]);
        $this->setPersonVersion($pValues[4]);
        return $this;
    }

}

File: Dddml.Wms.JavaCommon/src/org/dddml/wms/domain/MonthPlanMvoStateEventIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class MonthPlanMvoStateEventIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "monthPlanIdPersonalNameFirstName",
            "monthPlanIdPersonalNameLastName",
            "monthPlanIdYear",
            "monthPlanIdMonth",
            "personVersion",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "String",
            "Integer",
            "Integer",
            "Long",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private MonthPlanMvoStateEventId value;

    public MonthPlanMvoStateEventIdFlattenedDto()
    {
        this(new MonthPlanMvoStateEventId());
    }

    public MonthPlanMvoStateEventIdFlattenedDto(MonthPlanMvoStateEventId value)
    {
        this.value = value;
    }

    public MonthPlanMvoStateEventId toMonthPlanMvoStateEventId()
    {
        return this.value;
    }


    public String getMonthPlanIdPersonalNameFirstName()
    {
        return this.value.getMonthPlanId().getPersonalName().getFirstName();
    }

    public void setMonthPlanIdPersonalNameFirstName(String monthPlanIdPersonalNameFirstName)
    {
        this.value.getMonthPlanId().getPersonalName().setFirstName(monthPlanIdPersonalNameFirstName);
    }

    public String getMonthPlanIdPersonalNameLastName()
    {
        return this.value.getMonthPlanId().getPersonalName().getLastName();
    }

    public void setMonthPlanIdPersonalNameLastName(String monthPlanIdPersonalNameLastName)
    {
        this.value.getMonthPlanId().getPersonalName().setLastName(monthPlanIdPersonalNameLastName);
    }

    public Integer getMonthPlanIdYear()
    {
        return this.value.getMonthPlanId().getYear();
    }

    public void setMonthPlanIdYear(Integer monthPlanIdYear)
    {
        this.value.getMonthPlanId().setYear(monthPlanIdYear);
    }

    public Integer getMonthPlanIdMonth()
    {
        return this.value.getMonthPlanId().getMonth();
    }

    public void setMonthPlanIdMonth(Integer monthPlanIdMonth)
    {
        this.value.getMonthPlanId().setMonth(monthPlanIdMonth);
    }

    public Long getPersonVersion()
    {
        return this.value.getPersonVersion();
    }

    public void setPersonVersion(Long personVersion)
    {
        this.value.setPersonVersion(personVersion);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        MonthPlanMvoStateEventIdFlattenedDto other = (MonthPlanMvoStateEventIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

